easybuild/easyconfigs/f/FSL/FSL-6.0.2-foss-2018b-Python-2.7.15.eb

```diff
@@ -0,0 +1,78 @@
+# This build still relies on the following libraries (or part of them) shipped with FSL:
+#    GDCHART, libprob, libcprob, cprob, newran, newmat
+# Attempts to use externally built libraries failed. Worth trying again in the future...
+#
+# Some tools like icmp are missing since they are built with the fslpython_install.sh script
+# which locally installs Miniconda and creates a virtual environment. This is too messy and
+# an alternative solution is necessary to install them.
+# https://github.com/easybuilders/easybuild-easyconfigs/issues/7899
+# https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FslInstallation#imcp.2Fimglob.2Fimmv_errors_after_installation
+
+name = 'FSL'
+version = '6.0.2'
+local_pyver = '2.7.15'
+versionsuffix = '-Python-%s' % local_pyver
+
+homepage = 'https://www.fmrib.ox.ac.uk/fsl/'
+description = """FSL is a comprehensive library of analysis tools for FMRI, MRI and DTI brain imaging data."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+toolchainopts = {'cstd': 'c++11'}
+
+source_urls = ["https://www.fmrib.ox.ac.uk/fsldownloads/"]
+sources = ['%(namelower)s-%(version)s-sources.tar.gz']
+patches = [
+    'FSL-6.0.2_Makefile_fixes.patch',
+    'FSL-6.0.2_Build_extras.patch',
+    'FSL-6.0.1_Melodic-use-ifstream-good.patch',
+    'FSL-6.0.1_Fix_fsl_exec_script.patch',
+]
+checksums = [
+    'c118b351c6cedb441af7e1b9d194cf344505ff53b417063f697b86305a908afd',  # fsl-6.0.2-sources.tar.gz
+    '3a14d9c3ea75d90fa3e4beb5da3cd11a60c6859441acaece02b32ccab6aa278f',  # FSL-6.0.2_Makefile_fixes.patch
+    '2b7ade2146fadbba6c983370bb030ac4e2b11d8bf5259eacc8f3054d3d11003f',  # FSL-6.0.2_Build_extras.patch
+    'c07644fbd89cf9c70db5a1a8f4f2918ab5daeb60cdf0ce4ee2b91f8ae48173fa',  # FSL-6.0.1_Melodic-use-ifstream-good.patch
+    '4479aa501a7e6e9f1a0063e4ddb680836147ba08673d3fccb9e1db3c4d910672',  # FSL-6.0.1_Fix_fsl_exec_script.patch
+]
+
+builddependencies = [
+    ('Boost', '1.67.0'),
+]
+
+dependencies = [
+    ('libgd', '2.2.5'),
+    ('libxml2', '2.9.8'),
+    ('libxml++', '2.40.1'),
+    ('SQLite', '3.24.0'),
+    ('libpng', '1.6.34'),
+    ('Tk', '8.6.8'),
+    ('NLopt', '2.4.2'),
+    ('freeglut', '3.0.0'),
+    ('expat', '2.2.5'),
+    ('zlib', '1.2.11'),
+    ('VTK', '8.1.1', '-Python-%s' % local_pyver),
+    ('GSL', '2.5'),
+    ('Qwt', '6.1.4'),
+]
+
+# FSLDIR needs to be defined when running postinstall to get the correct shebang
+# https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=ind1910&L=FSL&P=R86209
+postinstallcmds = [(
+    'FSLDIR=%(installdir)s PATH=%(installdir)s/fsl/fslpython/bin:$PATH '
+    '%(installdir)s/fsl/etc/fslconf/post_install.sh -f %(installdir)s/fsl; '
+)]
+
+modextravars = {
+    'FSLOUTPUTTYPE': 'NIFTI_GZ',
+    'FSLMULTIFILEQUIT': 'TRUE',
+    'FSLTCLSH': 'tclsh',
+    'FSLWISH': 'wish8.6'
+}
+
+# Adding the bin from the virtualenv was the only way to get things working...
+modextrapaths = {
+    'PATH': 'fsl/fslpython/bin',
+    'PYTHONPATH': 'fsl/fslpython/envs/fslpython/lib/python3.7/site-packages/',
```

Comment:
More stuff for the easyblock?

This `virtualenv` stuff is making me eyes bleed though (but I think @vanzod also broke his teeth on it recently)

Reply:
I know, it is absolutely horrid. Despite assurances from the devels that 'this should not be needed', it was the only way to make things work. I think for the devs it simply works on their system, because they use a hard-coded she-bang to the python from the virtualenv. That didn't work for me, possibly because my prefix is quite a bit longer (and has more special characters) than their 'default' prefix.

Anyway, I decided I'd rather have a horrible, working solution, than no solution at all ;-)

I believe that @verdurin was planning to get in touch with the devs at some point to improve their installation procedures...

Regarding all requests for moving things to the EasyBlock: I agree that the EasyConfig is getting a lot of stuff in there that could be in an EasyBlock. Especially since FSL already has it's own, customized EasyBlock. However, right now, I really don't have time to move that, test it, etc (I have another project that needs to be finished end of this year...).

Considering that:
- FSL seems to change it's installation repeatedly anyway (configuring the build changed quite a bit between FSL-6.0.1 and FSL-6.0.2... see the changes in the Makefile_fixes), so I'm not sure if what we would put in today would still be relevant tomorrow...
- I understood some people have been asking for an FSL-6.0.2 PR
would it be an option to simply accept this somewhat cluttered EasyConfig? We can always 'beautify' it later... Or maybe one of the core EasyBuild maintainers could pick it up and do it - though I assume you're also all super busy =)